### ISSUE TITLE
ZeroMQ High Water Mark Sample Dropping

### ISSUE THREAD

**Issue by samwhiting** [gnuradio/gnuradio#6192]
### What happened?

When using the ZMQ PUB Sink and ZMQ SUB Source blocks, there is a parameter "High Watermark" that defaults to -1. In ZeroMQ, this sets the corresponding ZMQ_HWM socket option to a value of "0" which means infinite-sized buffers.

http://api.zeromq.org/2-1:zmq-setsockopt

An infinite-sized buffer is not always desired, so a user may specify a "High Watermark" value that in some way limits how much data (not sure how to convert this number into bytes or samples) can be stored in that buffer at any time. This seems like a good option - if a hardware source is producing data and sending it over ZMQ, maybe we want the ability to still back-pressure if the consuming flowgraph cannot keep up.

The actual default behavior of the ZMQ PUB socket is to drop samples (silently!) when this high watermark is reached. I would argue that if a GNURadio user is attempting to set a high-water mark, they want this back pressure behavior and should have an option to select that behavior through an input parameter. This can easily be done by setting the "ZMQ_XPUB_NODROP" socket option.

**tldr;**

**ZeroMQ PUB will drop samples without telling you if you use the high watermark feature.**

Example flowgraph sends a complex sinusoid over ZMQ PUB/ZUB with a high watermark of 1. The SUB side is intentionally throttled so it cannot keep up, samples get dropped.

![image](https://user-images.githubusercontent.com/23368909/192646345-d02b8c29-5857-4e13-a0a2-ac891d78fac0.png)

![image](https://user-images.githubusercontent.com/23368909/192646598-885b5cf2-4456-4dfb-a432-94ffadd81c00.png)

### System Information

OS: Ubuntu 20.04
GR Installation Method: Source

### GNU Radio Version

3.11-git (main)

### Specific Version

v3.11.0.0git-231-g463c3477

### Steps to Reproduce the Problem

Set a high watermark on PUB and SUB blocks, then send over more data than can be consumed on the SUB side. See example flowgraph above for a reference.
